### DISCUSSION TITLE
Are `packageManager` fields in all `package.json` files within a pnpm workspace required?

### DISCUSSION THREAD

**Discussion by ChristianIvicevic** [pnpm/pnpm#8853] [Q&A] [answered]
I am using corepack and have the `packageManager` field as `"pnpm@9.14.4"` in all `package.json` files, both the root file as well as the files of each package of the workspace. For housekeeping purposes I was considering to remove the field in all the files except the root `package.json` but I haven't found any documentation on whether these are required or the root config is being considered when selecting a version.

**Comment by zkochan** [comment, marked as answer]
It is enough to keep the `packageManager` field just in the root of the workspace for pnpm. But I don't know how other package managers will work in that case and whether they will tell you to use pnpm instead.

**Reply by ChristianIvicevic** [reply]
I see, thanks for clarification. Since `pnpm` is the standard for this project and i don't plan on migrating to other package managers it should mostly be fine and running commands from the folders inside should be a rare case as I'm using Turbo and thus invoking commands from the root anyways.
